tests/rptest/tests/full_disk_test.py

```diff
@@ -287,24 +288,42 @@ def observed_data_size(pred):
                                  if path.parts[0] == 'kafka')
             return pred(observed_total)
 
-        # write around 30 megabytes into the topic
-        produce_total_bytes(self.redpanda, self.topic, nbytes(30))
+        partition_count = 10
+
+        # We need at least 2 segments to be produced in order for the log eviction manager
+        # to be triggered.
+        num_segments = 2
+        for i in range(0, partition_count):
+            produce_until_segments(self.redpanda,
+                                   self.topic,
+                                   i,
+                                   num_segments,
+                                   acks=-1,
+                                   record_size=1024,
+                                   batch_size=1000)
+
+        # Calculate expected number of bytes based on partition count, number of segments, and segment size.
+        # Allow some fuzz factor to avoid timeouts.
+        fuzz_factor = 0.8
```

Comment:
The comment is confusing. Why do we need the fuzz factor? How does it help avoid timeouts? How the value is determined and why 0.8 is a good value? If we ever get a failure here the next person to look into it won't be able to understand anything unless they go through the same exercise you did. Please document what you've learnt to help the next person touching this code.

Reply:
FYI there is also this setting `log_segment_size_jitter_percent` which you might want to turn off for this test. Maybe this is why you had to include this fuzz factor?

Reply:
Nicolae is right, we should turn the segment size jitter off (although i thought it was off already for all ducktape tests)

Reply:
The previous test attempted to produce `30MB` of data while only asserting on `25MB`, hence the arbitrary `0.8` factor in my updated test as a way to give the test some tolerance. 

I'm happy to remove it though.

Can confirm `log_segment_size_jitter_percent` is [off by default for cluster configurations in ducktape](https://github.com/redpanda-data/redpanda/blob/f897aa1d234ab56cf5e97d657c8e6b9856e9f55d/tests/rptest/services/redpanda.py#L1276-L1279).

Reply:
Removed `fuzz_factor` and edited comments.